### PR TITLE
Add Parallel Cross Entropy

### PR DESCRIPTION
**Pull request by zhenglongjiepheonix**
# What does this PR do?

- [x] add parallel cross-entropy implementation
- [x] add parallel axis propagation rules for cross-entropy
- [x] add parallel cross-entropy replacement logic in pass
- [x] modify tests to trigger the critical path

### CODE REVIEW

**Review by michaelbenayoun — APPROVED** [review]
LGTM!
Is it for the default backend?

**Comment by zhenglongjiepheonix** [comment]
> LGTM! Is it for the default backend?

Nanotron backend will also use it because there is a `label_mask` input in Nanotron which is not compatible with transformers current implementation, the current implementation of shared cross entropy is compatible with transformers implementation, and it does not matter because cross entropy layer does not contain any backend-specific stuffs, it does not even contain a single parameter

**Comment by zhenglongjiepheonix** [comment]
If everything is good, can we merge this, I don't have write access to the main branch now :(
